### PR TITLE
Add geo.py module for geospatial operations

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR adds a `geo` utility module to perform geospatial transformations. Currently it has the capability to perform the following conversions:
- lon,lat,alt -> x,y,z
- x,y,z -> lon,lat,alt
- x,y,z -> east,north,up
- east,north,up -> x,y,z

It resolves the following issues:
- #24 
- #27 
- #28

### CODE REVIEW

**Comment by lsetiawan** [comment]
Check out `plh2xyz`. This will take geodetic to geocentric.

File: src/seagap/utilities/geo.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    return np.array(
+        [
+            [-sin_lon, -sin_lat * cos_lon, cos_lat * cos_lon],
+            [cos_lon, -sin_lat * sin_lon, cos_lat * sin_lon],
+            [0, cos_lat, sin_lat],
+        ]
+    )
+
+
+def geocentric2enu(x, y, z, lon_org, lat_org, alt_org):
```
Comment (carlosgjs) [inline src/seagap/utilities/geo.py:95]: add type annotations ....

Comment (carlosgjs) [reply src/seagap/utilities/geo.py:95]: also in enu2geocentric

Comment (lsetiawan) [inline src/seagap/utilities/geo.py:95]:
```suggestion
def geocentric2enu(x: float, y: float, z: float, lon_org: float, lat_org: float, alt_org: float):
```

**Pull request merged**